### PR TITLE
Recipe App: RoR group project

### PR DESCRIPTION
**Pull request by Bondok6**
# Hello there 🙋‍♂️,

### Thanks for taking the time to review our project.
<hr>

## What was done in this PR by @algerina & @Bondok6 🧾
 
- [x] Use Ruby gems as a software package system.
- [x]  Install Ruby on Rails framework.
- [x]  Understand Rails RESTful design and router.
- [x]  Use controllers to handle requests and render empty views.
- [x]  Understand Rails naming conventions.
- [x]  Use params from browser requests in a safe way.
- [x]  Use preprocessed HTML file with embedded Ruby code.
- [x]  Use layouts and templates for shared content.
- [x]  Use database migration files to maintain database schema.
- [x]  Use validations for models.
- [x]  Secure app from n+1 problems.
- [x]  Understand what ORM is.
- [x]  Write SQL queries with ActiveRecord.
- [x]  Set up associations between models.
- [x]  Build a web app that requires the user to log in.
- [x]  Describe the difference between authorization and authentication.
- [x]  Use Devise gem for authentication.
- [x]  Understand how sessions and cookies can support authentication.
- [x]  Limit access to web app resources based on authorization rules.
- [x] Use Gitflow
- [x] Add README.md

## To Run the test
```bash
rspec spec/views/
```

## Preview

https://user-images.githubusercontent.com/45231276/169404938-d17d8cba-d292-43c6-8d6f-eb5d92e0f30c.mp4

### CODE REVIEW

File: app/views/public_recipes/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+        <div class="col-6">
+          <p>Total Food Item: <span class="badge bg-danger"> <%= recipe.total_food_items %> </span></p> 
+          <p>Total Price: <span class="badge bg-danger"><%= recipe.total_price %> </span></p>
+        </div>
+      </div>
+    </div>
+  <%end%>
+
+</div>
```
Comment (GzimAsani) [inline app/views/public_recipes/index.html.erb]: - [x] Here the user should be able to click the public recipe which will then lead to  recipe details

File: app/views/recipes/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        <div class="col-3 text-center">
+        <a href="/recipes/<%= recipe.id %>" class="link-secondary fs-3"><%= recipe.name %></a>
+        <p><%= button_to "Delete", "recipes/#{recipe.id}" , method: :delete, class: 'btn btn-danger' %></p>
+        </div>
+        <div class="col-6">
+        <p><%=recipe.description%></p>
+        </div>
+      </div>
+    </div>
+  <%end%>
```
Comment (GzimAsani) [inline app/views/recipes/index.html.erb:19]: - [x] When another user checks for another public recipe it does not match the given wireframe ,kindly fix it, it looks kind of broken on the right side
Screenshot from 2022-05-19 23-28-16

File: app/views/shared/_navbar.html.erb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          <li class="nav-item">
+            <%= link_to 'Public Recipes', public_recipes_path, class: 'nav-link' %>
+          </li>
+
+          <li class="nav-item">
+            <%= link_to 'Edit Profile', edit_user_registration_path, class: 'nav-link' %>
+          </li>
+
+          <li class="nav-item">
+            <%= button_to 'Logout', destroy_user_session_path, method: :delete, class: 'nav-link btn btn-outline-danger' %>
```
Comment (GzimAsani) [inline app/views/shared/_navbar.html.erb]: - [x] When I click logout, it logs out but it does not update the page immediately, it lets the user know only after he clicked something else. KIndly improve this for a better user experience

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
+# This file is copied to spec/ when you run 'rails generate rspec:install'
+require 'spec_helper'
```
Comment (GzimAsani) [inline spec/rails_helper.rb:2]: - [x] Please add more tests for both unit and integration which cover more in-depth your app

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hi @Bondok6 and @algerina 
 
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Awesome job, I really liked your project :wave: 
 
 ## Required Changes 
- [ ]  _Check inline comments :arrow_double_down:_ 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by EricMbouwe — APPROVED** [review]
Hi @Bondok6 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**